src/main/java/org/broadinstitute/dsm/model/elasticsearch/ElasticSearch.java

```diff
@@ -1,161 +1,253 @@
-                    break;
-                case "proxies":
-                    builder.withProxies(GSON.fromJson(GSON.toJson(entry.getValue()), new TypeToken<List<Object>>() {}.getType()));
-                    break;
-                case "workflows":
-                    builder.withWorkFlows(GSON.fromJson(GSON.toJson(entry.getValue()), new TypeToken<List<Map<String, Object>>>() {}.getType()));
-                    break;
-                case "status":
-                    builder.withStatus(GSON.fromJson(GSON.toJson(entry.getValue()), String.class));
-                    break;
-                case "dsm":
-                    builder.withDsm(GSON.fromJson(GSON.toJson(entry.getValue()), new TypeToken<Map<String, Object>>() {}.getType()));
-                    break;
-                default:
-                    break;
-            }
-        }
-        return Optional.of(builder.build());
+    public long getTotalCount() {
+        return totalCount;
     }
 
-    public static class Builder {
-        private Optional<ESAddress> address = Optional.empty();
-        private Optional<List<Object>> medicalProviders = Optional.empty();
-        private Optional<List<Object>> invitations = Optional.empty();
-        private Optional<List<ESActivities>> activities = Optional.empty();
-        private Optional<Long> statusTimeStamp = Optional.empty();
-        private Optional<ESProfile> profile = Optional.empty();
-        private Optional<List<Object>> files = Optional.empty();
-        private Optional<List<Object>> proxies = Optional.empty();
-        private Optional<List<Map<String, String>>> workflows = Optional.empty();
-        private Optional<String> status = Optional.empty();
-        private Optional<Map<String, Object>> dsm = Optional.empty();
-
-        public Builder() {}
-
-        public Builder withAddress(ESAddress esAddress) {
-            this.address = Optional.ofNullable(esAddress);
-            return this;
-        }
+    public static Optional<ElasticSearchParticipantDto> parseSourceMap(Map<String, Object> sourceMap) {
+        if (sourceMap == null) return Optional.of(new ElasticSearchParticipantDto.Builder().build());
+        ElasticSearchParticipantDto elasticSearchParticipantDto = GSON.fromJson(GSON.toJson(sourceMap), ElasticSearchParticipantDto.class);
+        return Optional.of(elasticSearchParticipantDto);
+    }
 
-        public Builder withMedicalProviders(List<Object> medicalProviders) {
-            this.medicalProviders = Optional.ofNullable(medicalProviders);
-            return this;
+    public List<ElasticSearchParticipantDto> parseSourceMaps(SearchHit[] searchHits) {
+        if (Objects.isNull(searchHits)) return Collections.emptyList();
+        List<ElasticSearchParticipantDto> result = new ArrayList<>();
+        String ddp = getDdpFromSearchHit(Arrays.stream(searchHits).findFirst().orElse(null));
```

Comment:
that is the ddp string we display at the participant list, yes? I wonder if we should display the "display_name" from the ddp_instance table for that

Reply:
Yes, I don't know if it worth to make `ElasticSearch` class depend on some DB related classes because of ddp name?

Reply:
no, I think for now we leave it as it is. Maybe down the road we can add something to have the nice display_name, but also not in the ElasticSearch class, I think more about displaySettings bringing the display_name over and UI just using it …